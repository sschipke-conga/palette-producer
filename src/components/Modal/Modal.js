import React from 'react';
import {Route, NavLink, Link} from 'react-router-dom';
import ReactModal from 'react-modal';
import LoginForm from '../LoginForm/LoginForm';
import SignupForm from '../SignupForm/SignupForm';
import './Modal.scss';

export const Modal = () => {
  return (
    <>
      <Link to='/'>
        <div className="modal-background">
          <div className='modal-backdrop'></div>
        </div>
      </Link>
      <div className='modal-div'>
        <ul className='tabs'>
          <NavLink to='/login' className='modal-link'>
            <li className='modal-tab'>Log in</li>
          </NavLink>
          <NavLink to='/signup' className='modal-link'>
            <li className='modal-tab'>Sign up</li>
          </NavLink>
        </ul>
        <Route exact path='/login' render={() => <LoginForm />}
        />
        <Route exact path='/signup' render={() => <SignupForm />}
        />
      </div>
    </>
  )
}

export default Modal;