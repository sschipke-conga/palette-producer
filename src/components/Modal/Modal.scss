

.modal-backdrop {
  background: rgb(191, 191, 191);
  width: 100vw;
  height: 110vh;
  position: fixed;
  top: 50%;
  left: 50%;
  background-position: center top;
  background-size: cover;
  opacity: 0.9;
  transform: translate3d(-50%, -50%, 0);
  object-fit: cover;
  z-index: 1;
}

.modal-div {
  position: fixed;
  z-index: 100;
  top: 50%;
  left: 50%;
  transform: translate3d(-50%, -50%, 0);
  display: flex;
  flex-direction: column;
  background-color: white;
  border-radius: 6px 6px 4px 4px;
  border: none;
  box-shadow: 0 2px 10px rgba(0, 0, 0, 0.301);
  min-width: 20rem;
  max-width: 60rem;
  // border-bottom:
  width: 500px;
  height: 600px;
  // background: linear-gradient(to top left, #01C369, white, white, white, white);

  li {
    width: 250px;
    list-style-type: none;
    text-align: center;
    font-size: 30px;
    font-weight: 500;
    padding: 10px 0;
  }

  ul {
    width: 100%;
    border-radius: 4px 4px 0 0;
  }
}

.modal-link {
  text-decoration: none;
  border-radius: 4px 4px 0 0;
  color: white;
  transition: .15s;

  &:hover {
    background: rgb(0, 0, 0);
    transition: .2s;
  }
}

.active {
  color: rgb(27, 27, 27);
  background: white;
  border-radius: 4px 4px 0 0;

  &:hover {
    background: white;
  }
}

.tabs {
  display: flex;
  justify-content: space-evenly;
  width: 100%;
  background: rgb(41, 41, 41);
}

form {
display: flex;
align-items: center;
justify-content: space-between;
flex-direction: column;
height: 50%;
padding-top: 20px;
width: 100%;
  input {
    border:none;
  }
}

.error-message {
  color:red;
}