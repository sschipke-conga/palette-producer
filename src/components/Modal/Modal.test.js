import React from "react";
import Modal from "./Modal";
import { shallow } from "enzyme";

describe("Modal", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Modal />);
  });
  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
