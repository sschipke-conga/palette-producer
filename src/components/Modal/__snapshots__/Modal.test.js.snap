// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match the snapshot 1`] = `
<Fragment>
  <Link
    to="/"
  >
    <div
      className="modal-background"
    >
      <div
        className="modal-backdrop"
      />
    </div>
  </Link>
  <div
    className="modal-div"
  >
    <ul
      className="tabs"
    >
      <NavLink
        className="modal-link"
        to="/login"
      >
        <li
          className="modal-tab"
        >
          Log in
        </li>
      </NavLink>
      <NavLink
        className="modal-link"
        to="/signup"
      >
        <li
          className="modal-tab"
        >
          Sign up
        </li>
      </NavLink>
    </ul>
    <Route
      exact={true}
      path="/login"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/signup"
      render={[Function]}
    />
  </div>
</Fragment>
`;
